### DISCUSSION TITLE
ESM only package warning always showing up when importing into JSX

### DISCUSSION THREAD

**Discussion by 39digits** [remix-run/remix#5189] [Q&A]
### What version of Remix are you using?

1.6.7

### Steps to Reproduce

1. Create a fresh Remix app using `npx create-remix@latest`
2. Select the defaults - "Just the basics", "Remix App Server", and "TypeScript"
3. Install mdx-bundler (for example) using `npm install --save mdx-bundler esbuild`
4. Open `app/routes/index.tsx` and add an import from the mdx-bundler client library to the top of the file `import { getMDXComponent } from 'mdx-bundler/client';` - we don't need to use it anywhere just import.
5. Run the app with `npm run dev` - all is good
6. Shutdown the app
7. Rename to `index.jsx` and restart and the ESM only package warning `mdx-bundler is possibly an ESM only package and should be bundled with "serverDependenciesToBundle in remix.config.js.` shows up on the server output.
8. Aside: Adding the package to `serverDependenciesToBundle` does not affect the warning showing up - it always shows for JSX components.

Alternatively: 
1. Use create-remix but instead target JavaScript instead of TypeScript
2. Install mdx-bundler (for example) using `npm install --save mdx-bundler esbuild`
4. Open `app/routes/index.jsx` and add an import from the mdx-bundler client library to the top of the file `import { getMDXComponent } from 'mdx-bundler/client';` - we don't need to use it anywhere just import.
5. Run the app with `npm run dev` - warning appears

### Expected Behavior

Client-friendly imports should not show the warning on JSX components.

I would expect the warning to not show for JSX components in the same way it does not show up for TSX components - unless I'm misunderstanding something?

### Actual Behavior

The warning `mdx-bundler is possibly an ESM only package and should be bundled with "serverDependenciesToBundle in remix.config.js` shows in the server console for JSX components.  The warning does not show when using TSX components.

**Comment by matei-s** [comment]
I stumbled upon this problem a while ago and I thought I just have to accept it, but the solution is actually pretty simple: add `mdx-bundler/client` to `serverDependenciesToBundle` (keep `mdx-bundler`).


`remix.config.js`
```js
/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  ...
  serverDependenciesToBundle: [
    'mdx-bundler',
    'mdx-bundler/client', //   <--
    ...
  ],
  ...
}
```

Note: I am currently using `remix@1.18.1`
